autrainer/criterions/classification.py

```diff
@@ -0,0 +1,168 @@
+from typing import TYPE_CHECKING, Dict, List
+
+import pandas as pd
+import torch
+
+
+if TYPE_CHECKING:  # pragma: no cover
+    from autrainer.datasets import AbstractDataset
+
+
+class CrossEntropyLoss(torch.nn.CrossEntropyLoss):
+    def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        """Wrapper for `torch.nn.CrossEntropyLoss.forward`.
+
+        Converts the targets to `long` if it is a 1D tensor.
+
+        Args:
+            x: Batched model outputs.
+            y: Targets.
+
+        Returns:
+            Loss.
+        """
+        if y.ndim == 1:
+            y = y.long()
+        return super().forward(x, y)
+
+
+class BalancedCrossEntropyLoss(CrossEntropyLoss):
+    def setup(self, data: "AbstractDataset") -> None:
+        """Calculate balanced weights for the dataset based on the target
+        frequency in the training set.
+
+        Args:
+            data: Instance of the dataset.
+        """
+        frequency = (
+            data.df_train[data.target_column]
+            .map(data.target_transform)
+            .value_counts()
+            .sort_index()
+            .values
+        )
+        weight = torch.tensor(1 / frequency, dtype=torch.float32)
```

Comment:
1 / frequency could potentially lead to division by zero for missing train samples

```suggestion
        frequency = (
            data.df_train[data.target_column]
            .map(data.target_transform)
            .value_counts()
            .sort_index()
            .values
        )
        if (frequency == 0).any():
    raise ValueError("Balanced weighting failed: at least one class/label has zero frequency.")
        weight = torch.tensor(1 / frequency, dtype=torch.float32)
```